scripts/cmake/vcpkg_configure_make.cmake

```
@@ -477,20 +590,34 @@ function(vcpkg_configure_make)
             unset(ENV{PKG_CONFIG_PATH})
         endif()
         unset(BACKUP_ENV_PKG_CONFIG_PATH_${_buildtype})
+
+        if(_csc_ADD_BIN_TO_PATH)
+            set(ENV{PATH} "${PATH_BACKUP}")
+        endif()
+        # Restore environment (config dependent)
+        foreach(ENV_VAR ${_csc_CONFIG_DEPENDENT_ENVIRONMENT})
+            if(BACKUP_CONFIG_${ENV_VAR})
+                set(ENV{${ENV_VAR}} "${BACKUP_CONFIG_${ENV_VAR}}")
+            else()
+                unset(ENV{${ENV_VAR}})
+            endif()
+        endforeach()
+    endforeach()
+
+    # Restore environment
+    foreach(_prefix IN LISTS FLAGPREFIXES)
+        if(${prefix}_FLAGS_BACKUP)
+            set(ENV{${prefix}FLAGS} ${${_prefix}_FLAGS_BACKUP})
+        else()
+            unset(ENV{${prefix}FLAGS})
+        endif()

```

```suggestion
        _vcpkg_restore_env_variable(${prefix}FLAGS)
```
along with a corresponding change above to back them up as `${prefix}FLAGS`?